kolibri/core/lessons/viewsets.py

```diff
@@ -97,3 +101,26 @@ def annotate_queryset(self, queryset):
         return annotate_array_aggregate(
             queryset, lesson_assignment_collections="lesson_assignments__collection"
         )
+
+    @action(detail=False)
+    def size(self, request, **kwargs):
+        lessons = self.get_queryset()
+        if lessons:
+            lessons_nodes_list = [{lesson.id: lesson.resources} for lesson in lessons]
+            lessons_sizes_dict = {}
+
+            for lesson_node_item in lessons_nodes_list:
+                current_lesson_id = list(lesson_node_item.keys())[0]
+                current_lesson_resources_list = list(lesson_node_item.values())[0]
+                resource_nodes = []
```

Comment:
`total_file_size` takes a queryset of ContentNodes, so the simplest thing to do here is to create one from the lesson resources:
```
resources_nodes = ContentNode.objects.filter(id__in=[r["contentnode_id"] for r in lesson.resources])
```